### PR TITLE
Increase favicon size relative to the circle radius

### PR DESCRIPTION
**Pull request by Bilie**
Hi

This is aPR for issue [#204](https://github.com/mozilla/lightbeam-we/issues/204)
Looking forward to your feedback!

### CODE REVIEW

**Comment by princiya** [comment]
Thanks for the PR😊. At the moment i am travelling and have poor internet.
Will get back in 2 days.

**Comment by jonathanKingston** [comment]
updates-favicon-lightbeam

One side note is we don't want to request the image again which fixing this might lead to.
Instead pick a max favicon size for `convertURIToImageData` and scale it to the right size when it is being drawn in `drawFavicon`.

File: src/js/viz.js [hunk 1]
```diff
@@ -212,16 +207,28 @@ const viz = {
     });
   },
 
-  async drawFavicon(node, x, y) {
-    const offset = this.getSquare().offset,
+  async drawFavicon(node, x, y, radius) {
```
Comment (princiya) [inline src/js/viz.js]: @Bilie thanks for this PR. However, I am not convinced about how moving the code to `drawFavicon` function changes anything? Please correct me if I am missing anything obvious.
Moreover, now we have lost the caching functionality and on each call to `drawFavicon`, `imageCanvas` is being set.
Here is the [PR](https://github.com/mozilla/lightbeam-we/pull/201/files) where we implemented the caching functionality. With the moving of the code into the `drawFavicon` function, we now have come back to the non-caching version.

Comment (jonathanKingston) [reply src/js/viz.js]: From what I can tell this PR caches the network request which is great however In think we can save ourselves from re-rendering the favicon on every load twice.  Currently you are returning the canvas object then scaling it down. I personally think it makes more sense for this to be done inside `convertURIToImageData` as @princiya is mentioning and return the image data of a 64px icon perhaps, then use this to scale down to the right size on the main canvas.

So I think something like the following code can be in  `convertURIToImageData`:
```
    const imageContext = node.imageCanvas.getContext('2d');
    const side = this.getSquare(radius).side * this.scale;

    node.imageCanvas.width = side;
    node.imageCanvas.height = side;
    imageContext.fillStyle = this.canvasColor;
    imageContext.fillRect(0, 0, this.maxSideSize, this.maxSideSize);
    imageContext.drawImage(node.image, 0, 0, this.maxSideSize, this.maxSideSize);

   return imageContext.getImageData(0, 0, this.maxSideSize, this.maxSideSize);
```

Then in `drawIcon` just scale that icon down to the right size.

**Comment by Bilie** [comment]
Hi, thank you very much for all your feedback! I just updated the PR, even though I am not sure this is what was expected. I have tried scaling down the image within `putImageData`, but then the image was just cropped. The only way I found scaling the image is when it is actually drawn with `drawImage`. The image now should be loaded once, but then the image is redrawn on the canvas in the correct size every time nodes are updated.

**Comment by jonathanKingston** [comment]
This appears not to have the perf issues before thanks @Bilie!

**Pull request merged**